astronet: failed provided unit test "kepler_spline_test"

I have all the package required as instructed, and verified by "pip list". When I ran this:

bazel test astronet/... light_curve_util/... third_party/...

everything passed but the last one:

//third_party/kepler_spline:kepler_spline_test                           FAILED in 9.9s

Here's more detail from the log:

exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
/<my python virtual env dir>/lib/python2.7/site-packages/astropy/config/configuration.py:541: ConfigurationMissingWarning: Configuration defaults will be used due to OSError:Could not find unix home directory to search for astropy config dir on None
  warn(ConfigurationMissingWarning(msg))
Running tests under Python 2.7.10: /<my python virtual env dir>/bin/python
..F.
======================================================================
FAIL: testKeplerSplineError (__main__.KeplerSplineTest)
testKeplerSplineError (__main__.KeplerSplineTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_kelvinchan/c98b1016b78a661b9b19951be7e24696/bazel-sandbox/2476825790040633851/execroot/__main__/bazel-out/darwin-fastbuild/bin/third_party/kepler_spline/kepler_spline_test.runfiles/__main__/third_party/kepler_spline/kepler_spline_test.py", line 70, in testKeplerSplineError
    kepler_spline.kepler_spline(time, flux, bkspace=0.5)
AssertionError: SplineError not raised

----------------------------------------------------------------------
Ran 4 tests in 6.309s

FAILED (failures=1)

--- a/research/astronet/light_curve_util/util.py
+++ b/research/astronet/light_curve_util/util.py
@@ -18,8 +18,6 @@ from __future__ import absolute_import
 from __future__ import division
 from __future__ import print_function
 
-import collections
-
 import numpy as np
 from six.moves import range  # pylint:disable=redefined-builtin
 
@@ -51,21 +49,21 @@ def split(all_time, all_flux, gap_width=0.75):
   piecewise defined (e.g. split by quarter breaks or gaps in the in the data).
 
   Args:
-    all_time: Numpy array or list of numpy arrays; each is a sequence of time
-        values.
-    all_flux: Numpy array or list of numpy arrays; each is a sequence of flux
-        values of the corresponding time array.
+    all_time: Numpy array or sequence of numpy arrays; each is a sequence of
+        time values.
+    all_flux: Numpy array or sequence of numpy arrays; each is a sequence of
+        flux values of the corresponding time array.
     gap_width: Minimum gap size (in time units) for a split.
 
   Returns:
     out_time: List of numpy arrays; the split time arrays.
     out_flux: List of numpy arrays; the split flux arrays.
   """
+  all_time = np.array(all_time)
+  all_flux = np.array(all_flux)
+
   # Handle single-segment inputs.
-  # We must use an explicit length test on all_time because implicit conversion
-  # to bool fails if all_time is a numpy array, and all_time.size is not defined
-  # if all_time is a list of numpy arrays.
-  if len(all_time) > 0 and not isinstance(all_time[0], collections.Iterable):  # pylint:disable=g-explicit-length-test
+  if all_time.ndim == 1:
     all_time = [all_time]
     all_flux = [all_flux]
 
@@ -90,10 +88,10 @@ def remove_events(all_time, all_flux, events, width_factor=1.0):
   curve (e.g. one that is split by quarter breaks or gaps in the in the data).
 
   Args:
-    all_time: Numpy array or list of numpy arrays; each is a sequence of time
-        values.
-    all_flux: Numpy array or list of numpy arrays; each is a sequence of flux
-        values of the corresponding time array.
+    all_time: Numpy array or sequence of numpy arrays; each is a sequence of
+        time values.
+    all_flux: Numpy array or sequence of numpy arrays; each is a sequence of
+        flux values of the corresponding time array.
     events: List of Event objects to remove.
     width_factor: Fractional multiplier of the duration of each event to remove.
 
@@ -103,11 +101,11 @@ def remove_events(all_time, all_flux, events, width_factor=1.0):
     output_flux: Numpy array or list of numpy arrays; the flux arrays with
         events removed.
   """
+  all_time = np.array(all_time)
+  all_flux = np.array(all_flux)
+
   # Handle single-segment inputs.
-  # We must use an explicit length test on all_time because implicit conversion
-  # to bool fails if all_time is a numpy array and all_time.size is not defined
-  # if all_time is a list of numpy arrays.
-  if len(all_time) > 0 and not isinstance(all_time[0], collections.Iterable):  # pylint:disable=g-explicit-length-test
+  if all_time.ndim == 1:
     all_time = [all_time]
     all_flux = [all_flux]
     single_segment = True
@@ -150,10 +148,10 @@ def interpolate_masked_spline(all_time, all_masked_time, all_masked_spline):
   interp_spline = []
   for time, masked_time, masked_spline in zip(
       all_time, all_masked_time, all_masked_spline):
-    if len(masked_time) > 0:  # pylint:disable=g-explicit-length-test
+    if masked_time.size:
       interp_spline.append(np.interp(time, masked_time, masked_spline))
     else:
-      interp_spline.append(np.full_like(time, np.nan))
+      interp_spline.append(np.array([np.nan] * len(time)))
   return interp_spline
 
 

--- a/research/astronet/light_curve_util/util.py
+++ b/research/astronet/light_curve_util/util.py
@@ -59,11 +59,8 @@ def split(all_time, all_flux, gap_width=0.75):
     out_time: List of numpy arrays; the split time arrays.
     out_flux: List of numpy arrays; the split flux arrays.
   """
-  all_time = np.array(all_time)
-  all_flux = np.array(all_flux)
-
   # Handle single-segment inputs.
-  if all_time.ndim == 1:
+  if isinstance(all_time, np.ndarray) and all_time.ndim == 1:
     all_time = [all_time]
     all_flux = [all_flux]
 
@@ -101,11 +98,8 @@ def remove_events(all_time, all_flux, events, width_factor=1.0):
     output_flux: Numpy array or list of numpy arrays; the flux arrays with
         events removed.
   """
-  all_time = np.array(all_time)
-  all_flux = np.array(all_flux)
-
   # Handle single-segment inputs.
-  if all_time.ndim == 1:
+  if isinstance(all_time, np.ndarray) and all_time.ndim == 1:
     all_time = [all_time]
     all_flux = [all_flux]
     single_segment = True

--- a/research/astronet/light_curve_util/util_test.py
+++ b/research/astronet/light_curve_util/util_test.py
@@ -65,43 +65,63 @@ class LightCurveUtilTest(absltest.TestCase):
     self.assertSequenceAlmostEqual(expected, tfold)
 
   def testSplit(self):
+    # Single segment.
+    all_time = np.concatenate([np.arange(0, 1, 0.1), np.arange(1.5, 2, 0.1)])
+    all_flux = np.ones(15)
+
+    # Gap width 0.5.
+    split_time, split_flux = util.split(all_time, all_flux, gap_width=0.5)
+    self.assertLen(split_time, 2)
+    self.assertLen(split_flux, 2)
+    self.assertSequenceAlmostEqual(np.arange(0, 1, 0.1), split_time[0])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[0])
+    self.assertSequenceAlmostEqual(np.arange(1.5, 2, 0.1), split_time[1])
+    self.assertSequenceAlmostEqual(np.ones(5), split_flux[1])
+
+    # Multi segment.
     all_time = [
         np.concatenate([
             np.arange(0, 1, 0.1),
             np.arange(1.5, 2, 0.1),
             np.arange(3, 4, 0.1)
-        ])
+        ]),
+        np.arange(4, 5, 0.1)
     ]
-    all_flux = [np.array([1] * 25)]
+    all_flux = [np.ones(25), np.ones(10)]
+
+    self.assertEqual(len(all_time), 2)
+    self.assertEqual(len(all_time[0]), 25)
+    self.assertEqual(len(all_time[1]), 10)
+
+    self.assertEqual(len(all_flux), 2)
+    self.assertEqual(len(all_flux[0]), 25)
+    self.assertEqual(len(all_flux[1]), 10)
 
     # Gap width 0.5.
     split_time, split_flux = util.split(all_time, all_flux, gap_width=0.5)
-    self.assertLen(split_time, 3)
-    self.assertLen(split_flux, 3)
-    self.assertSequenceAlmostEqual(
-        [0., 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9], split_time[0])
-    self.assertSequenceAlmostEqual([1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
-                                   split_flux[0])
-    self.assertSequenceAlmostEqual([1.5, 1.6, 1.7, 1.8, 1.9], split_time[1])
-    self.assertSequenceAlmostEqual([1, 1, 1, 1, 1], split_flux[1])
-    self.assertSequenceAlmostEqual(
-        [3., 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9], split_time[2])
-    self.assertSequenceAlmostEqual([1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
-                                   split_flux[2])
+    self.assertLen(split_time, 4)
+    self.assertLen(split_flux, 4)
+    self.assertSequenceAlmostEqual(np.arange(0, 1, 0.1), split_time[0])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[0])
+    self.assertSequenceAlmostEqual(np.arange(1.5, 2, 0.1), split_time[1])
+    self.assertSequenceAlmostEqual(np.ones(5), split_flux[1])
+    self.assertSequenceAlmostEqual(np.arange(3, 4, 0.1), split_time[2])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[2])
+    self.assertSequenceAlmostEqual(np.arange(4, 5, 0.1), split_time[3])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[3])
 
     # Gap width 1.0.
     split_time, split_flux = util.split(all_time, all_flux, gap_width=1)
-    self.assertLen(split_time, 2)
-    self.assertLen(split_flux, 2)
+    self.assertLen(split_time, 3)
+    self.assertLen(split_flux, 3)
     self.assertSequenceAlmostEqual([
         0., 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.5, 1.6, 1.7, 1.8, 1.9
     ], split_time[0])
-    self.assertSequenceAlmostEqual(
-        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], split_flux[0])
-    self.assertSequenceAlmostEqual(
-        [3., 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9], split_time[1])
-    self.assertSequenceAlmostEqual([1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
-                                   split_flux[1])
+    self.assertSequenceAlmostEqual(np.ones(15), split_flux[0])
+    self.assertSequenceAlmostEqual(np.arange(3, 4, 0.1), split_time[1])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[1])
+    self.assertSequenceAlmostEqual(np.arange(4, 5, 0.1), split_time[2])
+    self.assertSequenceAlmostEqual(np.ones(10), split_flux[2])
 
   def testRemoveEvents(self):
     time = np.arange(20, dtype=np.float)

--- a/research/astronet/light_curve_util/util_test.py
+++ b/research/astronet/light_curve_util/util_test.py
@@ -136,20 +136,23 @@ class LightCurveUtilTest(absltest.TestCase):
     all_time = [
         np.arange(0, 10, dtype=np.float),
         np.arange(10, 20, dtype=np.float),
+        np.arange(20, 30, dtype=np.float),
     ]
     all_masked_time = [
         np.array([0, 1, 2, 3, 8, 9], dtype=np.float),  # No 4, 5, 6, 7
         np.array([10, 11, 12, 13, 14, 15, 16], dtype=np.float),  # No 17, 18, 19
+        np.array([], dtype=np.float)
     ]
     all_masked_spline = [2 * t + 100 for t in all_masked_time]
 
     interp_spline = util.interpolate_masked_spline(all_time, all_masked_time,
                                                    all_masked_spline)
-    self.assertLen(interp_spline, 2)
+    self.assertLen(interp_spline, 3)
     self.assertSequenceAlmostEqual(
         [100, 102, 104, 106, 108, 110, 112, 114, 116, 118], interp_spline[0])
     self.assertSequenceAlmostEqual(
         [120, 122, 124, 126, 128, 130, 132, 132, 132, 132], interp_spline[1])
+    self.assertTrue(np.all(np.isnan(interp_spline[2])))
 
   def testCountTransitPoints(self):
     time = np.concatenate([

